packages/alea-frontend/components/nap/GradingInterface.tsx

```diff
@@ -413,6 +431,26 @@ function GradingItemDisplay({
     refreshGradingInfo();
   }, [homeworkId, questionId, studentId]);
 
+  useEffect(() => {
+    const fetchProblem = async () => {
+      if (!questionMap[questionId]) {
+        try {
+          const uriWithoutId = questionId.replace(/en\?.*$/, 'en');
+          const learningObject = await getProblemObject(mmtUrl,uriWithoutId);
+          const problemHtml = await getLearningObjectShtml(mmtUrl, learningObject);
```

Comment:
refactor out this (removing hash code, getting problem object, shtml, removing id, get problem) into a separate function